public_html/wp-content/mu-plugins/blocks/assets/src/sponsors/edit.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Displays sponsor block.
+ */
+import SponsorInspectorControls from './inspector-controls';
+import SponsorBlockControls from './block-controls';
+
+/**
+ WordPress dependencies.
+ **/
+const { withSelect } = wp.data;
+const { Component, Fragment } = wp.element;
+const apiFetch = wp.apiFetch;
+const { addQueryArgs } = wp.url;
+
+const MAX_PAGE = 100;
+
+class SponsorsEdit extends Component {
+
+	/**
+	 * Constructor for SponsorsEdit block.
+	 *
+	 * @param props
+	 */
+	constructor( props ) {
+		super( props );
+		this.state = {
+			allSponsorPosts : null,
+			allSponsorTerms : null,
+			isLoading       : null,
+		}
+	}
+
+	/**
+	 * Renders SponsorEdit component.
+	 */
+	render() {
+		const { mode } = this.props.attributes;
+
+		return (
+			<Fragment>
+				{
+					<SponsorBlockControls
+						{ ...this.props }
+					/>
+				}
+				<Fragment>
+					<SponsorInspectorControls {...this.props} />
+				</Fragment>
+			</Fragment>
+		)
+	}
+}
+
+/**
+ * API call for wcb_sponsor post type data. Fetches all sponsor and terms posts.
+ *
+ * @param select
+ * @param props
+ */
+const sponsorSelect = ( select, props ) => {
+
+	const sponsorQuery = {
+		orderby : 'title',
+		order   : 'asc',
+		per_page: MAX_PAGE,
+		_embed  : true,
+	};
+
+	const sponsorLevelQuery = {
+		orderby : 'id',
+		order: 'asc',
+		per_page: MAX_PAGE,
+		_embed: true
+	};
+
+	return {
+		sponsorPosts: apiFetch( { path: addQueryArgs( '/wp/v2/sponsors', sponsorQuery ) } ),
+		sponsorLevels: apiFetch( { path: addQueryArgs('/wp/v2/sponsor_level', sponsorLevelQuery ) } )
```

Comment:
Ah, this is a regression, thanks for pointing it out. I have fixed with a cleaner solution, so basically I am fetching all the posts in one API call ( since we need them anyway ) for sponsor and one for sponsor level. Then doing the filtering on client side. 
This approach prevents at least 1 API call per resource and have less code.

Reply:
Ah cool. I will take a look at that. It might be worth doing that for all the blocks. My original reasoning for using `withSelect` was that the query parameters could result in different sets of speakers, with different sort orders. But since we're currently retrieving all the posts anyway for the Select, you're right that we could just pull them all once, and then do the filtering and sorting in-browser.

Reply:
We decided that we will use `withSelect` instead of `apiFetch`, so resolving this conversation. Tracking for implementing `withSelect` is here https://github.com/WordPress/wordcamp.org/issues/65.